### PR TITLE
Fix: pbs_python fails to initialize

### PR DESCRIPTION
**Pull request by raviv1995**
#### Describe Bug or Feature
pbs_python fails to initialize because of a missing '/'.

#### Describe Your Change
Added the missing '/'
Refactoring some MoM & MoM-hook env related funcs.

#### Attach Test and Valgrind Logs/Output

th

### CODE REVIEW

File: src/resmom/mom_hook_func.c [hunk 1]
```diff
@@ -1535,9 +1535,21 @@ run_hook(hook *phook, unsigned int event_type, mom_hook_input_t *hook_input,
 			}
 		}
 		(void)win_alarm(phook->alarm, run_hook_alarm);
-		run_exit = wsystem(cmdline, pwdp->pw_userlogin);
+		char *env_string = NULL;
+		char **hook_env = NULL;
+		char *pbs_hook_conf = NULL;
+
+		if (pjob->ji_env != NULL) {
+			hook_env = dup_string_arr(pjob->ji_env);
```
Comment (npadole20) [inline src/resmom/mom_hook_func.c:1543]: What happens if dup_string_arr() fails? We should exit in that case.

**Review by npadole20 — CHANGES_REQUESTED** [review]
Please attach test logs.

**Comment by bayucan** [comment]
@raviv1995 : I think @npadole20 has a question for you...

**Comment by npadole20** [comment]
@bayucan  @raviv1995  has made the change at https://github.com/openpbs/openpbs/pull/1980/files#diff-f000d05f102c29b1985e578963befa7aR1517
@raviv1995  the changes look good to me. please attach test logs.

**Review by npadole20 — APPROVED** [review]

File: src/resmom/requests.c [hunk 2]
```diff
@@ -2791,7 +2791,7 @@ req_cpyfile(struct batch_request *preq)
 		 * but they would be inherited by the next request handler
 		 * and will cause interference
 		 */
-		init_envp();
+		free_string_array(pjob->ji_env);
```
Comment (subhasisb) [inline src/resmom/requests.c:2794]: why freeing ji_env out of the blue?

File: src/server/job_func.c [hunk 3]
```diff
@@ -838,6 +839,10 @@ job_purge(job *pjob)
 		reply_text(pjob->ji_preq, PBSE_INTERNAL, "job deleted");
 		pjob->ji_preq = NULL;
 	}
+
+	if (pjob->ji_env != NULL)
```
Comment (subhasisb) [inline src/server/job_func.c:843]: had u initialized ji_env to nULL in job_alloc()?

Comment (subhasisb) [inline src/server/job_func.c:843]: u dont need to check for NULL before calling free_string_array()

File: src/resmom/mom_main.c [hunk 4]
```diff
@@ -2780,67 +2781,65 @@ do_mom_action_script(int	ae,	/* index into action table */
...
 
-	init_envp();
-
 	/*
 	 **	Setup environment
 	 */
 	/* UID */
 	sprintf(buf, "%d", pjob->ji_qs.ji_un.ji_momt.ji_exuid);
-	bld_wenv_variables("UID", buf);
+	env_array = bld_wenv_variables("UID", buf, env_array);
```
Comment (subhasisb) [inline src/resmom/mom_main.c:2789]: check if bld_wenv_variables() can be deprecated in lieu of the mainstream bld_env_variables() function. Basically, we should try to reuse bld_env_variables() and modify that if it needs enhancement

**Review by subhasisb — CHANGES_REQUESTED** [review]

**Review by subhasisb — APPROVED** [review]

**Comment by raviv1995** [comment]
Thanks for the review @bayucan @npadole20 @subhasisb

**Pull request merged**